src/migrations/rebase.rs

```diff
@@ -0,0 +1,286 @@
+use std::collections::HashMap;
+use std::iter::FromIterator;
+use fs_err as fs;
+
+use std::ops::Index;
+use anyhow::{anyhow, Context as _};
+use colorful::Colorful;
+use indexmap::IndexMap;
+use crate::connect::Connection;
+use crate::migrations::{Context, create, db_migration, migrate, migration};
+use crate::migrations::create::{MigrationKey, MigrationToText};
+use crate::migrations::db_migration::{DBMigration, read_all};
+use crate::migrations::dev_mode::rebase_to_schema;
+use crate::migrations::extract::DatabaseMigration;
+use crate::migrations::grammar::parse_migration;
+use crate::migrations::migrate::apply_migrations_inner;
+use crate::migrations::migration::MigrationFile;
+use crate::print;
+
+#[derive(PartialEq)]
+enum RebaseMigrationKind {
+    Base,
+    Source,
+    Target,
+}
+
+struct RebaseMigration<'a> {
+    key: MigrationKey,
+    migration: &'a DBMigration,
+    parent: Option<&'a str>,
+    kind: RebaseMigrationKind
+}
+
+impl MigrationToText for RebaseMigration<'_> {
+    type StatementsIter<'a> = std::iter::Once<&'a String> where Self: 'a;
+
+    fn key(&self) -> &MigrationKey {
+        &self.key
+    }
+
+    fn parent(&self) -> anyhow::Result<&str> {
+        if let Some(parent) = self.parent {
+            return Ok(parent)
+        }
+
+        if self.migration.parent_names.is_empty() {
+            return Ok("initial");
+        }
+
+        Ok(&self.migration.parent_names[0])
```

Comment:
This shouldn't be needed. We are iterating over all migrations anyway, so our parent should be set correctly.

If it is not, that is a bug.